lightly_studio/src/lightly_studio/api/routes/api/caption.py

```diff
@@ -9,21 +9,23 @@
 
 from lightly_studio.api.routes.api.validators import Paginated, PaginatedWithCursor
 from lightly_studio.db_manager import SessionDep
-from lightly_studio.models.caption import CaptionsListView
+from lightly_studio.models.caption import SampleCaptionsListView
 from lightly_studio.resolvers import caption_resolver
-from lightly_studio.resolvers.caption_resolver import GetAllCaptionsResult
+from lightly_studio.resolvers.caption_resolver import (
+    GetAllCaptionsFromSampleResult,
+)
 
 captions_router = APIRouter(prefix="/datasets/{dataset_id}", tags=["captions"])
 
 
-@captions_router.get("/captions", response_model=CaptionsListView)
+@captions_router.get("/captions", response_model=SampleCaptionsListView)
 def read_captions(
     dataset_id: Annotated[UUID, Path(title="Dataset Id")],
     session: SessionDep,
     pagination: Annotated[PaginatedWithCursor, Depends()],
-) -> GetAllCaptionsResult:
+) -> GetAllCaptionsFromSampleResult:
     """Retrieve captions for a dataset."""
-    return caption_resolver.get_all(
```

Comment:
This is the only place we use `caption_resolver.get_all`. We can remove the function (or better said - modify it to fit the new needs).